uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestMailNotification.php

```diff
@@ -0,0 +1,391 @@
+      if ($leaveRequest->sickness_required_documents) {
+        $templateParameters['sicknessRequiredDocuments'] = $this->getSicknessRequiredDocuments();
+        $templateParameters['leaveRequiredDocuments'] =  explode(',', $leaveRequest->sickness_required_documents);
+      }
+    }
+
+    return $templateParameters;
+  }
+
+  /**
+   * get Email template to be used based on request type
+   *
+   * @param \CRM_HRLeaveAndAbsences_BAO_LeaveRequest $leaveRequest
+   *
+   * @return \CRM_Core_DAO_MessageTemplate
+   */
+  public function getTemplate(LeaveRequest $leaveRequest) {
+    $templateName = '';
+
+    switch ($leaveRequest->request_type) {
+      case LeaveRequest::REQUEST_TYPE_SICKNESS:
+        $templateName = 'CiviHR Sickness Record Notification';
+        break;
+      case LeaveRequest::REQUEST_TYPE_LEAVE:
+        $templateName = 'CiviHR Leave Request Notification';
+        break;
+      case LeaveRequest::REQUEST_TYPE_TOIL:
+        $templateName = 'CiviHR TOIL Request Notification';
+        break;
+    }
+
+    $params = ['msg_title' => $templateName, 'is_default' => 1];
+    $defaults = [];
+    $template = MessageTemplate::retrieve($params, $defaults);
+
+    return $template;
+  }
+
+  /**
+   * Gets the From Email address configured on the site
+   * Currently It gets the from email address and name of the contact linked to the
+   * default domain in civicrm_domain table.
+   *
+   * This logic will be changed later to fetch from L&A general settings
+   *
+   * @return string
+   */
+  private function getFromEmail() {
+    if (is_null($this->fromEmail)) {
+      $domainValues = [];
+      $domain = CRM_Core_BAO_Domain::getDomain();
+      $tokens = [
+        'domain' => ['name', 'email'],
+      ];
+
+      foreach ($tokens['domain'] as $token) {
+        $domainValues[$token] = CRM_Utils_Token::getDomainTokenReplacement($token, $domain);
+      }
+
+      $this->fromEmail = $domainValues['name'] . ' <' . $domainValues['email'] . '>';
+    }
+    return $this->fromEmail;
+  }
+
+  /**
+   * Returns an array of emails of eligible recipients for this leave Request Notification
+   *
+   * @param \CRM_HRLeaveAndAbsences_BAO_LeaveRequest $leaveRequest
+   *
+   * @return array
+   */
+  public function getRecipientEmails(LeaveRequest $leaveRequest) {
+    $leaveApprovers = $this->getLeaveApprovers($leaveRequest);
+
+    if (empty($leaveApprovers)) {
+      $leaveApprovers = $this->getNotificationReceivers($leaveRequest);
+    }
+
+    $contactEmail = [$leaveRequest->contact_id];
+    $allRecipients = array_merge($leaveApprovers, $contactEmail);
+
+    $result = civicrm_api3('Email', 'get', [
+      'contact_id' => ['IN' => $allRecipients],
+      'is_primary' => 1,
+      'api.Contact.get' => ['id' => '$value.contact_id', 'return' => ['display_name']],
+      'return' => ['email','contact_id']
+    ]);
+
+    return $result['values'];
+  }
+
+  /**
+   * Gets the contactID's of Leave approvers for the current leave request
```

Comment:
I think this will describe this method a bit better: "Gets the Contact ID's of the currently active Leave Approvers of the given Leave Request"